src/shared/widgets/rich-text/rich-text-widget.tsx

```diff
@@ -0,0 +1,70 @@
+import React, { useCallback, useEffect, useRef, useState } from "react";
+import { WidgetProps } from "react-jsonschema-form";
+import { getContentHeight, htmlToSlate, SlateEditor, slateToHtml, SlateToolbar } from "@concord-consortium/slate-editor";
+import { useRefState } from "../../hooks/use-ref-state";
+import "@concord-consortium/slate-editor/build/index.css";
+import "./rich-text-widget.scss";
+
+export const RichTextWidget = (props: WidgetProps) => {
+  const [value, valueRef, setValue] = useRefState(htmlToSlate(props.value || ""));
+  const [changeCount, setChangeCount] = useState(0);
+  const editorRef = useRef<any>();
+  const kExtraHeight = 30;
+  const kInitialHeight = 50;
+  const [height, setHeight] = useState(kInitialHeight);
+
+  const handleEditorRef = useCallback((editor: any | null) => {
+    editorRef.current = editor || undefined;
+    if (editor) {
+      // associate label with edit field
+      (editor as any).el.id = props.id;
+    };
+  }, [editorRef.current]);
+
+  const handleFocus = useCallback(() => {
+    props.onFocus(props.id, slateToHtml(valueRef.current));
+  }, [valueRef.current]);
```

Comment:
I'm not a big fan of `useCallback()` either and I'm certainly not using it for performance reasons. `handleFocus()` and `handleBlur()` are called asynchronously and this was the only way I could get them not to close on a stale state value. I may play around with it a bit more to see if there's another solution, but this was simplest solution I found that actually worked.

Reply:
I just tried this again and it turns out the the `useCallback()` isn't required. I thought I had tried that combination. The `useRefState()` is required, however, to prevent the stale closure. I've added a commit with the change.

Reply:
I don't think it's worth investigating, but in this case, it seems to me that this closure needs to happen somewhere in Slate Editor code. Maybe it doesn't update these event handlers once they're assigned or something like that? Now, we have regular functions in the top-level of the render function that should get recreated on every state update (including `value`).